Check undefined skip check slave lag schema change

cswingler opened the pull request:
The P parameter can be undefined in DSNs, resulting in the skip-check-slave-lag feature in pt-table-checksum puking trying to compare undefs.

This handles that check a little more safely.

This is exactly like #230 , but for `pt-online-schema-change`.

Commit:
Safely check for undefined values in --skip-check-slave-lag

Commit:
Attempting to line up with existing whitespace

Commit:
That method is named differently here

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.percona.com:443/pull/badge/not_signed)](https://cla-assistant.percona.com:443/percona/percona-toolkit?pullRequest=388) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.percona.com:443/percona/percona-toolkit?pullRequest=388) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: svetasmirnova<br/>:x: Chris Swingler<br/><hr/>**Chris Swingler** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.percona.com:443/check/percona/percona-toolkit?pullRequest=388) it.</sub>

hhgyu commented:
the issue still exists.

svetasmirnova reviewed (approved):
Thank you for your contribution.

Please sign CLA, so we can accept it.

svetasmirnova commented:
We may need to check if other tools require similar fix.

svetasmirnova committed:
PR-388 - Check undefined skip check slave lag schema change

- Merge branch '3.x' into HEAD
- Improve the fix, so it does not fail when $slave_to_skip->{P} is not
  defined while $slave->{dsn}->{P} is defined
- Improve the fix for pt-table-checksum (PR-230)
- Added test cases for this fix for pt-table-checksum and pt-osc
- Adjusted replica info repository options in test for pt-1760.t, so it
  does not break regression tests for pt-osc on 5.7
